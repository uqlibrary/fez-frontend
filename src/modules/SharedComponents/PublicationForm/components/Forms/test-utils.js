import { useForm } from 'react-hook-form';
import { render, WithReduxStore } from 'test-utils';
import React from 'react';
import { formValues } from '../PublicationForm';

export const setupWrapper = (Form, props) => {
    const Wrapper = () => {
        const {
            control,
            formState: { isSubmitting },
        } = useForm();

        // eslint-disable-next-line react/prop-types
        return <Form {...{ control, isSubmitting, ...props, formValues: formValues(props?.formValues || {}) }} />;
    };

    return render(
        <WithReduxStore>
            <Wrapper />
        </WithReduxStore>,
    );
};
