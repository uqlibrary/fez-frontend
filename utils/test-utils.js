/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { mui1theme } from 'config/theme';
import { Provider } from 'react-redux';

import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';

import { getStore } from '../src/config/store';
import Immutable from 'immutable';

import mediaQuery from 'css-mediaquery';

const domTestingLib = require('@testing-library/dom');
const reactTestingLib = require('@testing-library/react');
const mime = require('mime-types');

import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { waitFor } from '@testing-library/dom';
import preview from 'jest-preview';

export const AllTheProviders = props => {
    return (
        <MuiThemeProvider theme={mui1theme}>
            <LocalizationProvider dateAdapter={AdapterMoment}>{props.children}</LocalizationProvider>
        </MuiThemeProvider>
    );
};

AllTheProviders.propTypes = {
    children: PropTypes.node,
};

export const rtlRender = (ui, options) => {
    return {
        user: userEvent.setup(),
        ...render(ui, { wrapper: AllTheProviders, ...options }),
    };
};

export const WithRouter = ({ children, route = '/', initialEntries = [route] }) => {
    const routes = [{ path: route, element: children }];
    const router = createMemoryRouter(routes, { initialEntries: initialEntries });
    return <RouterProvider router={router} />;
};

export const withRedux = (initialState = Immutable.Map()) => WrappedComponent => {
    return <Provider store={getStore(initialState)}>{WrappedComponent}</Provider>;
};

export const WithReduxStore = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>
        <AllTheProviders>{children}</AllTheProviders>
    </Provider>
);

export const assertTooltipText = async (trigger, tooltipText) => {
    expect(trigger).toBeInTheDocument();
    await userEvent.hover(trigger);
    await waitFor(() => {
        expect(screen.getByRole('tooltip', { name: tooltipText, hidden: true })).toBeVisible();
    });
};

const extensionToMimeMap = {
    tiff: 'image/tiff',
    wav: 'audio/wave',
    zip: 'application/zip',
};
const mimeType = filename => extensionToMimeMap[filename.split('.').pop()];

export const defaultDatastream = {
    dsi_embargo_date: null,
    dsi_open_access: 1,
    dsi_label: '',
    dsi_copyright: null,
    dsi_state: 'A',
    dsi_size: 27932352,
};

export const createFezDatastreamInfoArray = (datastreams, pid = null, withPreview = true) => {
    let processed = datastreams.filter(datastream => typeof datastream === 'object' || typeof datastream === 'string');
    if (withPreview) {
        processed = [];
        datastreams.forEach(datastream => {
            processed.push(datastream);
            const filename = typeof datastream === 'object' ? datastream.dsi_dsid : datastream;
            const mimetype = mimeType(filename);
            // bail in case it's not derivable
            if (
                !filename.includes('.pdf') &&
                !mimetype.includes('image') &&
                !mimetype.includes('audio') &&
                !mimetype.includes('video')
            ) {
                return;
            }
            // eslint-disable-next-line no-nested-ternary
            const derivativeMimetype = mimetype.includes('audio')
                ? 'audio/mp3'
                : mimetype.includes('video')
                ? 'video/mp4'
                : 'image/jpg';

            (derivativeMimetype.includes('image') ? ['preview_', 'thumbnail_', 'web_'] : ['']).forEach(prefix => {
                processed.push({
                    dsi_dsid: `${prefix}${filename.split('.')[0]}_t.${derivativeMimetype.split('/').pop()}`,
                    dsi_mimetype: derivativeMimetype,
                });
            });
        });
    }

    return processed.map((datastream, index) => ({
        ...defaultDatastream,
        ...{
            dsi_id: index + 1,
            dsi_pid: pid,
            dsi_order: index + 1,
            dsi_mimetype: mime.lookup(typeof datastream === 'object' ? datastream.dsi_dsid : datastream),
        },
        ...(typeof datastream === 'object'
            ? datastream
            : {
                  dsi_dsid: datastream,
              }),
    }));
};

export const withDatastreams = (sources, datastreams, callback) =>
    datastreams.map(item => {
        const source = sources.find(datastream =>
            datastream.dsi_dsid.includes(
                item.dsi_dsid
                    .replace('_t.', '.')
                    .replace('preview_', '')
                    .replace('thumbnail_', '')
                    .replace('web_', '')
                    .split('.')[0],
            ),
        );
        const isDerivative = item.dsi_dsid !== source.dsi_dsid;
        callback(item, source, isDerivative);
    });

export const getDatastreamByFilename = (filename, datastreams) =>
    datastreams.find(datastream => datastream.dsi_dsid === filename);

export const createMatchMedia = width => {
    return query => ({
        matches: mediaQuery.match(query, { width }),
        /* istanbul ignore next */
        addListener: () => {},
        /* istanbul ignore next */
        removeListener: () => {},
    });
};

export const getFilenameExtension = filename => filename.split('.').pop();
export const getFilenameBasename = filename =>
    filename.replace(new RegExp(`/\.${getFilenameExtension(filename)}$/`), '');
export const addFilesToFileUploader = files => {
    const { screen, fireEvent } = reactTestingLib;
    // create a list of Files
    const fileList = files.map(file => {
        // if file it's a string, treat it as a filename
        if (typeof file === 'string') {
            return new File([getFilenameBasename(file)], file, { type: `image/${getFilenameExtension(file)}` });
        }
        // otherwise expect it to be a object with filename and mimeType keys
        return new File([getFilenameBasename(file.filename)], file.filename, { type: file.mimeType });
    });
    // drag and drop files
    fireEvent.drop(screen.getByTestId('fez-datastream-info-input'), {
        dataTransfer: {
            files: fileList,
            types: ['Files'],
        },
    });
};
export const setFileUploaderFilesToClosedAccess = async (files, timeout = 500) => {
    const { fireEvent } = reactTestingLib;
    // set all files to closed access
    for (const file of files) {
        const index = files.indexOf(file);
        await waitFor(() => screen.getByText(new RegExp(getFilenameBasename(file))), { timeout });
        fireEvent.mouseDown(screen.getByTestId(`dsi-open-access-${index}-select`));
        fireEvent.click(screen.getByRole('option', { name: 'Closed Access' }));
    }
};

export const assertEnabled = (resolver, ...items) => items.forEach(item => expect(resolver(item)).toBeEnabled());
export const assertDisabled = (resolver, ...items) => items.forEach(item => expect(resolver(item)).toBeDisabled());

module.exports = {
    ...domTestingLib,
    ...reactTestingLib,
    rtlRender,
    withRedux,
    AllTheProviders,
    WithReduxStore,
    assertTooltipText,
    WithRouter,
    createFezDatastreamInfoArray,
    getDatastreamByFilename,
    withDatastreams,
    createMatchMedia,
    preview,
    userEvent,
    assertEnabled,
    assertDisabled,
    getFilenameExtension,
    getFilenameBasename,
    addFilesToFileUploader,
    setFileUploaderFilesToClosedAccess,
};
