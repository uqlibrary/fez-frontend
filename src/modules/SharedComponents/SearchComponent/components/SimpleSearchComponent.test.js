import React from 'react';
import { SimpleSearchComponent } from './SimpleSearchComponent';
import { rtlRender, fireEvent, createEvent } from 'test-utils';

jest.mock('config/general', () => ({
    ...jest.requireActual('config/general'),
    MAX_PUBLIC_SEARCH_TEXT_LENGTH: 20,
}));

function setup(testProps = {}) {
    const props = {
        onSearchTextChange: jest.fn(),
        ...testProps,
    };

    return rtlRender(<SimpleSearchComponent {...props} />);
}

describe('SimpleSearchComponent', () => {
    it('should render default view', () => {
        const { container } = setup();
        expect(container).toMatchSnapshot();
    });

    it('should render mobile view', () => {
        const { container, getByRole } = setup({
            showMobileSearchButton: true,
            isInHeader: true,
            classes: {
                mobileHeader: 'mobileHeaderTest',
            },
        });
        fireEvent.click(getByRole('button', { name: 'Click to search eSpace' }));
        expect(container.querySelector('.mobileHeaderTest')).toBeInTheDocument();
    });

    it('should show prefix icon when in header', () => {
        const { container } = setup({
            isInHeader: true,
            showPrefixIcon: true,
            classes: {
                inHeader: 'headerClass',
                searchIconPrefix: 'searchIconPrefix',
            },
        });
        expect(container).toMatchSnapshot();
    });

    it('should render show prefix icon in the search box', () => {
        const { container } = setup({ showPrefixIcon: true });
        expect(container).toMatchSnapshot();
    });

    it('should render with a class "header" for use in AppBar', () => {
        const { container } = setup({ isInHeader: true });
        expect(container).toMatchSnapshot();

        const form = container.querySelector('form');
        const submitEvent = createEvent.submit(form);
        fireEvent(form, submitEvent);
        expect(submitEvent.defaultPrevented).toBe(true);
    });

    it('should set search value from prop', () => {
        const { container } = setup({ showAdvancedSearchButton: true, searchText: 'i feel lucky' });
        expect(container).toMatchSnapshot();
    });

    it('should update search text field', () => {
        const testFn = jest.fn();
        const { getByTestId } = setup({ onSearchTextChange: testFn });
        fireEvent.change(getByTestId('simple-search-input'), { target: { value: 'new search value' } });
        expect(testFn).toHaveBeenCalledWith('new search value');
    });

    it("should not submit search if ENTER wasn't pressed", () => {
        const testMethod = jest.fn();
        const { getByTestId } = setup({ searchText: 'new search value', onSearch: testMethod });
        fireEvent.keyDown(getByTestId('simple-search-input'), { key: 'Escape', code: 27 });
        expect(testMethod).not.toHaveBeenCalled();
    });

    it('should submit search if search text is not null and ENTER is pressed', () => {
        const searchFn = jest.fn();
        const { getByTestId } = setup({
            searchText: 'new search value',
            onSearch: searchFn,
        });
        fireEvent.keyDown(getByTestId('simple-search-input'), { key: 'Enter', code: 13 });
        expect(searchFn).toBeCalled();
    });

    it('should render with default onSearch callback', () => {
        const { container, getByTestId } = setup({
            searchText: 'new search value',
        });
        fireEvent.keyDown(getByTestId('simple-search-input'), { key: 'Enter', code: 13 });
        expect(container).toMatchSnapshot();
    });

    it('should ignore empty search string', () => {
        const searchFn = jest.fn();
        const { getByRole } = setup({
            searchText: ' ',
            onSearch: searchFn,
        });
        fireEvent.click(getByRole('button', { name: 'Click to search eSpace' }));
        expect(searchFn).not.toBeCalled();
    });

    it('should toggle search mode', () => {
        const testToggleFn = jest.fn();
        const { getByRole } = setup({
            showAdvancedSearchButton: true,
            onToggleSearchMode: testToggleFn,
        });
        fireEvent.click(getByRole('button', { name: 'Click to switch to Advanced search' }));
        expect(testToggleFn).toHaveBeenCalled();
    });

    it('should render with default toggle search mode callback', () => {
        const { container, getByRole } = setup({
            showAdvancedSearchButton: true,
        });
        fireEvent.click(getByRole('button', { name: 'Click to switch to Advanced search' }));
        expect(container).toBeInTheDocument();
    });

    it('should toggle mobile search', () => {
        const { getByTestId, getByRole } = setup({
            showMobileSearchButton: true,
            isInHeader: true,
        });
        fireEvent.click(getByRole('button', { name: 'Click to search eSpace' }));
        fireEvent.click(getByTestId('CloseIcon'));
        expect(getByTestId('SearchIcon')).toBeInTheDocument();
    });

    it('searchTextValidationMessage() should return a message for being too long', () => {
        const { getByText } = setup({ searchText: 'this is way way too long' });
        expect(getByText('Must be 20 characters or less')).toBeInTheDocument();
    });
});
