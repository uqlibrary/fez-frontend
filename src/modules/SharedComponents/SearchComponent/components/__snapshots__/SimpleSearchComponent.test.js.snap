// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleSearchComponent should render default view 1`] = `
<div>
  <form
    style="margin: 8px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="simple-search"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard css-h9aup-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            data-testid="simple-search-label"
            for="simple-search-input"
            id="simple-search-label"
          >
            Search eSpace
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1vg8b0p-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              aria-label="Enter your search query to search eSpace and then press Enter"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch css-c63i49-MuiInputBase-input-MuiInput-input"
              data-analyticsid="simple-search-input"
              data-testid="simple-search-input"
              id="simple-search-input"
              label="Search eSpace"
              placeholder="Enter your search query to search all fields in eSpace"
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto css-pr9bjb-MuiGrid-root"
      >
        <button
          aria-label="Click to search eSpace"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1duotrm-MuiButtonBase-root-MuiButton-root"
          data-analyticsid="simple-search-button"
          data-testid="simple-search-button"
          id="simple-search-button"
          tabindex="0"
          type="button"
        >
          Search
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto css-pr9bjb-MuiGrid-root"
      >
        <button
          aria-label="Click to switch to Advanced search"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedDefault MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorDefault MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedDefault MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorDefault MuiButton-fullWidth advancedButton css-7qz7dl-MuiButtonBase-root-MuiButton-root"
          data-analyticsid="show-advanced-search"
          data-testid="show-advanced-search"
          id="show-advanced-search"
          tabindex="0"
          type="button"
        >
          Advanced search
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`SimpleSearchComponent should render show prefix icon in the search box 1`] = `
<div>
  <form
    style="margin: 8px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="simple-search"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard css-h9aup-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            data-testid="simple-search-label"
            for="simple-search-input"
            id="simple-search-label"
          >
            Search eSpace
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1vg8b0p-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              aria-label="Enter your search query to search eSpace and then press Enter"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch css-c63i49-MuiInputBase-input-MuiInput-input"
              data-analyticsid="simple-search-input"
              data-testid="simple-search-input"
              id="simple-search-input"
              label="Search eSpace"
              placeholder="Enter your search query to search all fields in eSpace"
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto css-pr9bjb-MuiGrid-root"
      >
        <button
          aria-label="Click to search eSpace"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1duotrm-MuiButtonBase-root-MuiButton-root"
          data-analyticsid="simple-search-button"
          data-testid="simple-search-button"
          id="simple-search-button"
          tabindex="0"
          type="button"
        >
          Search
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto css-pr9bjb-MuiGrid-root"
      >
        <button
          aria-label="Click to switch to Advanced search"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedDefault MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorDefault MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedDefault MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorDefault MuiButton-fullWidth advancedButton css-7qz7dl-MuiButtonBase-root-MuiButton-root"
          data-analyticsid="show-advanced-search"
          data-testid="show-advanced-search"
          id="show-advanced-search"
          tabindex="0"
          type="button"
        >
          Advanced search
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`SimpleSearchComponent should render with a class "header" for use in AppBar 1`] = `
<div>
  <form
    style="margin: 8px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap css-44sczy-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-8moi20-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="simple-search"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-b1iuur-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              aria-label=""
              autocomplete="search"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch css-c63i49-MuiInputBase-input-MuiInput-input"
              data-analyticsid="simple-search-input"
              data-testid="simple-search-input"
              id="simple-search-input"
              label=""
              placeholder="Search eSpace"
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-10q9jzs"
    >
      <button
        aria-label="Click to search eSpace"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-bc0omg-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gmqj7h-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
          />
        </svg>
      </button>
    </div>
  </form>
</div>
`;

exports[`SimpleSearchComponent should render with default onSearch callback 1`] = `
<div>
  <form
    style="margin: 8px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="simple-search"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-5afocb-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            data-testid="simple-search-label"
            for="simple-search-input"
            id="simple-search-label"
          >
            Search eSpace
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1vg8b0p-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              aria-label="Enter your search query to search eSpace and then press Enter"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch css-c63i49-MuiInputBase-input-MuiInput-input"
              data-analyticsid="simple-search-input"
              data-testid="simple-search-input"
              id="simple-search-input"
              label="Search eSpace"
              placeholder="Enter your search query to search all fields in eSpace"
              type="search"
              value="new search value"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto css-pr9bjb-MuiGrid-root"
      >
        <button
          aria-label="Click to search eSpace"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1duotrm-MuiButtonBase-root-MuiButton-root"
          data-analyticsid="simple-search-button"
          data-testid="simple-search-button"
          id="simple-search-button"
          tabindex="0"
          type="button"
        >
          Search
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto css-pr9bjb-MuiGrid-root"
      >
        <button
          aria-label="Click to switch to Advanced search"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedDefault MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorDefault MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedDefault MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorDefault MuiButton-fullWidth advancedButton css-7qz7dl-MuiButtonBase-root-MuiButton-root"
          data-analyticsid="show-advanced-search"
          data-testid="show-advanced-search"
          id="show-advanced-search"
          tabindex="0"
          type="button"
        >
          Advanced search
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`SimpleSearchComponent should set search value from prop 1`] = `
<div>
  <form
    style="margin: 8px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="simple-search"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-5afocb-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            data-testid="simple-search-label"
            for="simple-search-input"
            id="simple-search-label"
          >
            Search eSpace
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1vg8b0p-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              aria-label="Enter your search query to search eSpace and then press Enter"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch css-c63i49-MuiInputBase-input-MuiInput-input"
              data-analyticsid="simple-search-input"
              data-testid="simple-search-input"
              id="simple-search-input"
              label="Search eSpace"
              placeholder="Enter your search query to search all fields in eSpace"
              type="search"
              value="i feel lucky"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto css-pr9bjb-MuiGrid-root"
      >
        <button
          aria-label="Click to search eSpace"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1duotrm-MuiButtonBase-root-MuiButton-root"
          data-analyticsid="simple-search-button"
          data-testid="simple-search-button"
          id="simple-search-button"
          tabindex="0"
          type="button"
        >
          Search
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto css-pr9bjb-MuiGrid-root"
      >
        <button
          aria-label="Click to switch to Advanced search"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedDefault MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorDefault MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedDefault MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorDefault MuiButton-fullWidth advancedButton css-7qz7dl-MuiButtonBase-root-MuiButton-root"
          data-analyticsid="show-advanced-search"
          data-testid="show-advanced-search"
          id="show-advanced-search"
          tabindex="0"
          type="button"
        >
          Advanced search
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`SimpleSearchComponent should show prefix icon when in header 1`] = `
<div>
  <form
    style="margin: 8px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap headerClass css-44sczy-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1g2tota-MuiGrid-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium searchIconPrefix css-iu4k2w-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
          />
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-8moi20-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="simple-search"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-b1iuur-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              aria-label=""
              autocomplete="search"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch css-c63i49-MuiInputBase-input-MuiInput-input"
              data-analyticsid="simple-search-input"
              data-testid="simple-search-input"
              id="simple-search-input"
              label=""
              placeholder="Search eSpace"
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-10q9jzs"
    >
      <button
        aria-label="Click to search eSpace"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-bc0omg-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gmqj7h-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
          />
        </svg>
      </button>
    </div>
  </form>
</div>
`;
